src/plasma/plasma_manager.cc

```diff
@@ -1346,10 +1349,21 @@ void process_object_notification(event_loop *loop,
                                  void *context,
                                  int events) {
   PlasmaManagerState *state = (PlasmaManagerState *) context;
-  ObjectInfo object_info;
   /* Read the notification from Plasma. */
-  int error =
-      read_bytes(client_sock, (uint8_t *) &object_info, sizeof(object_info));
+  int64_t size;
```

Comment:
Yes, I was thinking about this, the only reason why I didn't do it is that the error handling is slightly different in 2 of these 3 cases. I'll change it and decouple the error handling by returning a status code.